fulp_modules/features/antagonists/bloodsuckers/code/rulesets/monsterhunter_event.dm

```diff
@@ -42,31 +51,26 @@
 			living_players -= player
 
 /datum/dynamic_ruleset/midround/monsterhunter/proc/generate_monsters(amount)
-	var/list/possible_monsters = list(/datum/antagonist/bloodsucker,
-	/datum/antagonist/heretic,
-	/datum/antagonist/changeling)
 	for(var/i in 1 to amount)
 		var/mob/living/monster = pick(living_players)
+		var/datum/antagonist/profession = pick_weight(GLOB.monster_antagonist_types)
+		if(!monster.mind.add_antag_datum(profession))
+			continue
```

Comment:
this iteration will be counted in the loop, creating 1 less monster than we need everytime the monster can't be given the antag

Reply:
it adds to the amount to add if it fails